src/main/java/Main.java

```diff
@@ -0,0 +1,174 @@
+
+    public static class Resources {
+        String name;
+        Destructor destructor;
+
+        public Resources(String name, Destructor destructor) {
+            this.name = name;
+            this.destructor = destructor;
+        }
+    }
+
+    public static void main(String[] args) throws ConfigError, FieldConvertError, JMException {
+        CommonFactory factory;
+        ConcurrentLinkedDeque<Resources> resources = new ConcurrentLinkedDeque<>();
+        try {
+            Runtime.getRuntime().addShutdownHook(new Thread(
+                    () -> resources.descendingIterator().forEachRemaining((resource) -> {
+                        log.debug("Destroying resource: " + resource.name);
+                        try {
+                            resource.destructor.close();
+                        } catch (Exception e) {
+                            log.error("Failed to destroy resource: " + resource.name);
+                        }
+                    })
+            ));
+        } catch (Exception e) {
+            log.error("Uncaught exception. Shutting down");
+            System.exit(1);
+            throw new RuntimeException("System.exit returned normally, while it was supposed to halt JVM.");
+        }
+
+        try {
+            factory = CommonFactory.createFromArguments(args);
+            resources.add(new Resources("factory", factory::close));
+        } catch (Exception e) {
+            factory = new CommonFactory();
+            log.error("Failed to create common factory from args", e);
+        }
+
+        JsonMapper mapper = JsonMapper.builder().build();
+
+        Settings settings = factory.getCustomConfiguration(Settings.class, mapper);
+        MessageRouter<EventBatch> eventRouter = factory.getEventBatchRouter();
+        MessageRouter<MessageGroupBatch> messageRouter = factory.getMessageRouterMessageGroupBatch();
+        GrpcRouter grpcRouter = factory.getGrpcRouter();
+
+        run(settings, messageRouter, eventRouter, grpcRouter, resources);
+
+    }
+
+    public static void run(Settings settings, MessageRouter<MessageGroupBatch> messageRouter, MessageRouter<EventBatch> eventRouter,
+                           GrpcRouter grpcRouter, ConcurrentLinkedDeque<Resources> resources) throws ConfigError, FieldConvertError, JMException {
+
+        SessionFixBean sessionFixBean = new SessionFixBean();
+        sessionFixBean.createConfig();    //create config file for FIX
+        String sessionAlias = new SessionFixBean().getSessionAlias();
+
+        ConnectionID connectionID = ConnectionID.newBuilder().setSessionAlias(sessionAlias).build();
+
+        Event curEvent = MessageRouterUtils.storeEvent(eventRouter, Event.start(), null);
+        curEvent.name("FIX client " + sessionAlias + Instant.now());
+        curEvent.type("Microservice");
+        String rootEventId = curEvent.getId();
+
+        FixClient fixClient = new FixClient(new SessionSettings("src/main/resources/acceptor/acceptor.cfg"),
```

Comment:
There won't be `src/main/resources...` folder in built application. We should store config in a temporary file instead